sql/lease.go

```diff
@@ -151,11 +152,13 @@ func (s LeaseStore) Acquire(txn *client.Txn, tableID ID, minVersion DescriptorVe
 		}
 		return nil
 	})
```

Comment:
it's not equivalent. `*roachpb.Error` carries information which can get stripped by this cycle-cast.
The reason I suggested doing it is because it feels sketchy to pass a `*roachpb.Error` created by a txn closure up into the result of another, pending, closure. The inner closure runs to completion, so it shouldn't be able to pass a retry error from the inner transaction out, but with the advent of transaction retry options and the like I think it should be made clear.

Maybe `s.db.Txn` should simply sanitize the returned `pErr` instead of doing it at each call site?